### PR TITLE
Cleanup Helm RBAC

### PR DESCRIPTION
**Pull request by chrisob**
Master branch changes related to #11234.

What started as a bug fix ended up as a general cleanup of Helm `ClusterRole` rules (biggest cleanup is in Kiali's chart).

### CODE REVIEW

**Comment by chrisob** [comment]
@sdake Glad to help, RBAC apiVersions updated to v1.

Just wondering, these failed CI tests aren't due to my changes, right? :thinking:

**Comment by sdake** [comment]
@chrisob tewst-integatoin-kubernetes, istio-integ-k8s-tets.sh and istio-pilot-multicluster-e2e.sh are busted.

as for pilot-auth-v1alph3-v2 - unknown if this is because of your PR or not.  I have not seen this gate failure signature prior, but that doesn't mean it doesn't exist :)  I have rerun the workflow.

File: install/kubernetes/helm/subcharts/mixer/templates/clusterrole.yaml [hunk 1]
```diff
@@ -18,10 +18,7 @@ rules:
 - apiGroups: [""]
   resources: ["configmaps", "endpoints", "pods", "services", "namespaces", "secrets", "replicationcontrollers"]
   verbs: ["get", "list", "watch"]
-- apiGroups: ["extensions"]
-  resources: ["replicasets"]
```
Comment (sdake) [inline install/kubernetes/helm/subcharts/mixer/templates/clusterrole.yaml]: replicasets not needed for mixer?

**Comment by sdake** [comment]
if the PR fails to pass `required` circleci jobs, you might consider rebasing on master.  circleci does not automatically rebase before running jobs.

Cheers
-steve

File: install/kubernetes/helm/subcharts/mixer/templates/clusterrole.yaml (hunk 1 above)
Comment (chrisob) [reply install/kubernetes/helm/subcharts/mixer/templates/clusterrole.yaml]: @sdake 
Both the `extensions` and `apps` apiGroups contain a `replicasets` resource, so I merged this rule with the one below it (see `- apiGroups: ["extensions", "apps"]`).

**Comment by chrisob** [comment]
@morvencao look okay to you?

**Comment by morvencao** [comment]
@chrisob A try to rebase on `master` may fix the CI tests issues.

**Pull request merged**